core/core_integrations/messenger.js

```diff
@@ -0,0 +1,198 @@
+var fs = require('fs'),
+    path = require('path'),
+    express = require('express'),
+    bodyParser = require('body-parser'),
+    request = require('request'),
+    https = require('https'),
+    shim = require('../shim.js'),
+    http = null,
+    api = null,
+
+    getNameForId = function(id, callback) {
+        if (!exports.config.users) {
+            exports.config.users = {};
+        }
+
+        if (exports.config.users[id]) {
+            return callback(exports.config.users[id].name);
+        }
+
+        request({
+            url: 'https://graph.facebook.com/v2.6/' + id,
+            qs: {access_token: exports.config.token, fields: 'first_name,last_name'},
+            method: 'GET'
+        },
+        function(error, response, body) {
+            if (error) {
+                return callback('UnknownUser');
+            }
+            body = JSON.parse(body);
+            if (!exports.config.users[id]) {
+                exports.config.users[id] = {};
+            }
+            exports.config.users[id].name = body.first_name + ' ' + body.last_name;
+            callback(exports.config.users[id].name);
+        });
+    },
+
+    sentenceSplitter = function(message) {
+        var spl = message.split('\n');
+
+        for (var i = 0; i < spl.length; i++) {
+            var strs = [];
+            while (spl[i].length > 0) {
+                var str = spl[i].substr(0, 320);
+                var ind = str.lastIndexOf(' ');
+                if (ind >= 0 && str.length !== spl[i].length) {
+                    str = str.substr(0, Math.min(str.length, ind + 1));
+                }
+
+                strs.push(str);
+                spl[i] = spl[i].substr(str.length);
+            }
+            spl.splice(i, 1);
+            for (var j = 0; j < strs.length; j++) {
+                spl.splice(i + j, 0, strs[j]);
+            }
+            i += strs.length;
+        }
+
+        for (var i = 0; i < spl.length; i++) {
+            var curr = spl[i];
+            for (var j = i + 1; j < spl.length; j++) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: ["j" is already defined](https://www.codacy.com/app/theprojectshop/Kassy/file/2626852437/issues/source?bid=3282987&fileBranchId=3316939#l62)